latch/ldata/path.py

```diff
@@ -0,0 +1,171 @@
+                ldataResolvePathData(argPath: $argPath) {
+                    finalLinkTarget {
+                        childLdataTreeEdges(filter: { child: { removed: { equalTo: false } } }) {
+                            nodes {
+                                child {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"argPath": self.path},
+        )["ldataResolvePathData"]
+
+        if data is None:
+            raise ValueError(f"No directory found at path: {self.path}")
+
+        for node in data["finalLinkTarget"]["childLdataTreeEdges"]["nodes"]:
+            yield urljoins(self.path, node["child"]["name"])
+
+    def rmr(self) -> None:
+        execute(
+            gql.gql("""
+            mutation LDataRmr($nodeId: BigInt!) {
+                ldataRmr(input: { argNodeId: $nodeId }) {
+                    clientMutationId
+                }
+            }
+            """),
+            {"nodeId": self.node_id},
+        )
+
+    def copy(self, dst: Union["LPath", str]) -> None:
+        remote_copy(self.path, str(dst))
+
+    def upload(self, src: Path, progress=Progress.tasks, verbose=False) -> None:
+        upload(src, self.path, progress, verbose)
+
+    def download(
+        self, dst: Optional[Path] = None, progress=Progress.tasks, verbose=False
+    ) -> Path:
+        if dst is None:
+            dir = Path(".") / "downloads" / str(uuid4())
+            dir.mkdir(parents=True, exist_ok=True)
+            dst = dir / self.name
+
+        download(self.path, dst, progress, verbose, confirm_overwrite=False)
+        return dst
+
+    @property
+    def perms(self) -> LDataPerms:
+        return get_node_perms(self.node_id)
+
+    def share_with(self, email: str, perm_level: PermLevel) -> None:
+        resp = post(
+            url=urljoin(NUCLEUS_URL, "/ldata/send-share-email"),
+            json={
+                "node_id": self.node_id,
+                "perm_level": str(perm_level),
+                "receiver_email": email,
+            },
+            headers={"Authorization": get_auth_header()},
+        )
+        resp.raise_for_status()
+
+    def _toggle_share_link(self, enable: bool) -> None:
+        execute(
+            gql.gql("""
+            mutation LDataShare($nodeId: BigInt!, $value: Boolean!) {
+                ldataShare(input: { argNodeId: $nodeId, argValue: $value }) {
+                    clientMutationId
+                }
+            }
+            """),
+            {"nodeId": self.node_id, "value": enable},
+        )
+
+    def enable_share_link(self) -> None:
+        self._toggle_share_link(True)
+
+    def disable_share_link(self) -> None:
+        self._toggle_share_link(False)
+
+    def __str__(self) -> str:
+        return self.path
+
+    def __truediv__(self, other: Union[Path, str]) -> "LPath":
```

Comment:
```suggestion
    def __truediv__(self, other: object) -> "LPath":
```

+ check `isinstance(other, (LPath, str))` and *return (don't throw)* `NotImplemented` if false.